cmd/age/age.go

```diff
@@ -174,19 +174,22 @@ func main() {
 		defer f.Close()
 		out = f
 	} else if terminal.IsTerminal(int(os.Stdout.Fd())) {
-		if armorFlag {
-			// If the output will go to a TTY, and it will be armored, buffer it
-			// up so it doesn't get in the way of typing the input.
+		if name != "-" {
+			if decryptFlag {
+				// TODO: buffer the output and check it's printable.
+			} else if !armorFlag  {
+				// If the output wouldn't be armored, refuse to send binary to
+				// the terminal unless explicitly requested with "-o -".
+				logFatalf("Error: refusing to output binary to the terminal.\n" +
+					`Did you mean to use -a/--armor? Force with "-o -".`)
+			}
+		}
+		if terminal.IsTerminal(int(os.Stdin.Fd())) {
```

Comment:
```suggestion
		if in == os.Stdin && terminal.IsTerminal(int(os.Stdin.Fd())) {
```

We don't want to buffer output for file inputs, right?

Reply:
Of course! How did I hide so many problems is this little patch 😳

Reply:
Hehe, it's messy logic :) Thank you for bringing it up!